lib/base-compiler.ts

```diff
@@ -969,25 +970,26 @@ export class BaseCompiler implements ICompiler {
             .filter(link => link) as string[];
     }
 
-    getSharedLibraryPaths(libraries: CompileChildLibraries[]): string[] {
+    getSharedLibraryPaths(libraries: CompileChildLibraries[], dirPath?: string): string[] {
         return libraries
             .map(selectedLib => {
                 const foundVersion = this.findLibVersion(selectedLib);
                 if (!foundVersion) return [];
 
                 const paths = [...foundVersion.libpath];
-                if (this.buildenvsetup && !this.buildenvsetup.extractAllToRoot) {
-                    paths.push(`/app/${selectedLib.id}/lib`);
+                if (this.buildenvsetup && !this.buildenvsetup.extractAllToRoot && dirPath) {
+                    paths.push(path.join(dirPath, selectedLib.id, 'lib'));
```

Comment:
nice to see a hardcoded `/app` going (which obviously wouldn't have worked for folks not sandboxing)